source/data-modeling/persistence-configuration.txt

```diff
@@ -0,0 +1,293 @@
+named ``analytics``:
+
+.. literalinclude:: /includes/data-modeling/persistence-configuration.rb
+   :language: ruby
+   :start-after: start store_in example
+   :end-before: end store_in example
+
+The ``store_in`` macro can also accept a lambda function. This is useful if you
+want to define a persistence context with values that cannot use a constant string. 
+
+You might want to use this pattern in a multi-tenant application,
+where multiple users share common access to an application. By using
+a lambda, you can define a persistence context based on the current thread so
+that users cannot access each others' data.
+
+The following example stores documents in a database determined by the 
+current thread:
+
+.. literalinclude:: /includes/data-modeling/persistence-configuration.rb
+   :language: ruby
+   :start-after: start store_in lambda example
+   :end-before: end store_in lambda example
+
+.. _mongoid-set-runtime-persistence-options:
+
+Runtime Persistence Options
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can use the ``with()`` method on a model class or instance to change
+a model's persistence configuration for a group of operations during runtime.
+
+Call the ``with()`` method on a model class or instance and pass options 
+that define a persistence context. The ``with()`` method accepts the following 
+arguments:
+
+- ``options_or_context``: A customizable set of options or a 
+  ``Mongoid::PersistenceContext`` object.
+
+- ``options``: A customizable set of options.
+
+Then, use a block to define the operations that you want to execute in the 
+specified context. The context that you define only exists while the code
+in the block runs.
+
+By default, {+odm+} stores documents for the ``Band`` class in a collection called
+``bands``. The following code example uses the ``with()`` method to temporarily
+use a different client, database, and collection to perform operations on the 
+``Band`` class's documents:
+
+.. literalinclude:: /includes/data-modeling/persistence-configuration.rb
+   :language: ruby
+   :start-after: start with() example
+   :end-before: end with() example
+
+.. note:: Define Clients
+
+   In the preceding example, you must define the ``tertiary`` cluster under
+   ``clients`` in your ``mongoid.yml`` file.
+
+.. important:: Block Scope
+
+   You must call the ``with()`` method with a block.
+   This is because {+odm+} uses the options you pass to the method to create
+   a new client in the background. A block defines the scope of this client
+   so it can be closed and its resources freed.
+
+You can also pass the ``with()`` method configuration options for read or write 
+operations. The configurations apply only to the specified type of operation.
+
+The following example uses the ``with()`` method to specify the use of the 
+secondary node for all read operations within the block. Write operations in 
+the block still run on the default context.
```

Comment:
*All* operations there use the new context, even write operations. However, since the read configuration is the only difference between the default and new persistence contexts, a write will behave the same on this new context as it would on the default context.

```suggestion
secondary node for all read operations within the block.
```